### DISCUSSION TITLE
decompilation of C# metadata without switching to nvim-lspconfig?!

### DISCUSSION THREAD

**Discussion by amirongit** [neoclide/coc.nvim#4917] [Q&A]
hey, I have my csharp lsp up & running using coc.nvim with this configuration in my coc-settings

```json
    "languageserver": {
        "csharp-ls": {
            "command": "csharp-ls",
            "filetypes": [
                "cs"
            ],
            "rootPatterns": [
                "*.csproj",
                ".git/",
                "*.sln"
            ]
        }
    }
```

the problem is that I can't use "go to definition" for third party libraries (well, I can; but I see an empty buffer).
I'm new to csharp & I think the buffer needs to be decompiled first (in order to see the content)
there is a vim plugin which does this; but their documentation uses lspconfig.
link: https://github.com/Decodetalkers/csharpls-extended-lsp.nvim?tab=readme-ov-file#usage
I don't know how to set this up with neovim. is it possible to somehow re write the lua code in the usage section to call the setup method on csharpls?
thanks in advance.

**Comment by amirongit** [comment]
so would it be possible to override the handler for textDocument/definition in certain filetypes?

**Comment by amirongit** [comment]
also, there is some information in csharpls' repo itself. could be useful?

link: https://github.com/razzmatazz/csharp-language-server?tab=readme-ov-file#decompile-for-your-editor--with-the-example-of-neovim

**Comment by fannheyward** [comment]
You need to create an extension of coc.nvim to decompilation the metadata, similar as vscode-csharp-ls, https://github.com/vytautassurvila/vscode-csharp-ls/blob/dd73a1733b2a68fb64512a7bdcf4bd2e03ce6d5b/src/cSharpLsServer.ts#L197-L209. Made a `csharp/metadata` request to csharp-ls,  coc.nvim also provides `workspace.registerTextDocumentContentProvider` for this.

You're setting up csharp-ls with `languageserver`, but `csharp/metadata` is off-spec, an extension is needed.

**Reply by fannheyward** [reply]
You can run `:echo CocRequestAsync('languageserver.csharp-ls', 'csharp/metadata', {'textDocument': {'uri': 'file:///path/to/your/file'}})` to make a test.

**Comment by amirongit** [comment]
so there MUST be an extension \\:
thank you for answering
